libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -33,20 +35,22 @@ Compass::compass_cal_update()
     if (is_calibrating()) {
         _cal_has_run = true;
         return;
-    } else if (_cal_has_run && auto_reboot()) {
+    } else if (_cal_has_run && _auto_reboot()) {
         hal.scheduler->delay(1000);
         hal.scheduler->reboot(false);
     }
 }
 
 bool
-Compass::start_calibration(uint8_t i, bool retry, bool autosave, float delay, bool autoreboot)
+Compass::_start_calibration(uint8_t i, bool retry, float delay)
 {
     if (!healthy(i)) {
         return false;
     }
-    memset(_reports_sent,0,sizeof(_reports_sent));
-    if (!is_calibrating() && delay > 0.5f) {
+    if (!use_for_yaw(i)) {
+        return false;
```

Comment:
Added thought on it. I'm really opposed to that for two reasons, thats not terribly clear to a MAVLink user, and requires the user to explicitly pick in a GCS which way they want to be running the calibration. I think restricting the calibration to only calibrate compasses that are going to be used is a very reasonable interface/design restriction. It is much easier to describe the documentation if the compass is always calibrated only if it will be used.